exporter/exporterhelper/tracehelper.go

```diff
@@ -116,3 +134,108 @@ func pushTraceDataWithSpan(next PushTraceData, spanName string) PushTraceData {
 		return droppedSpans, err
 	}
 }
+
+type otlpTraceExporter struct {
+	exporterFullName string
+	dataPusher       otlpTraceDataPusher
+	shutdown         Shutdown
+}
+
+var _ exporter.OTLPTraceExporter = (*otlpTraceExporter)(nil)
+
+func (te *otlpTraceExporter) Start(host component.Host) error {
+	return nil
+}
+
+func (te *otlpTraceExporter) ConsumeOTLPTrace(
+	ctx context.Context,
+	td consumerdata.OTLPTraceData,
+) error {
+	exporterCtx := observability.ContextWithExporterName(ctx, te.exporterFullName)
+	_, err := te.dataPusher(exporterCtx, td)
+	return err
+}
+
+// Shutdown stops the exporter and is invoked during shutdown.
+func (te *otlpTraceExporter) Shutdown() error {
+	return te.shutdown()
+}
+
+// NewOTLPTraceExporter creates an OTLPTraceExporter that can record metrics and can wrap
+// every request with a Span.
+func NewOTLPTraceExporter(
+	config configmodels.Exporter,
+	dataPusher otlpTraceDataPusher,
+	options ...ExporterOption,
+) (exporter.OTLPTraceExporter, error) {
+
+	if config == nil {
+		return nil, errNilConfig
+	}
+
+	if dataPusher == nil {
+		return nil, errNilPushTraceData
+	}
+
+	opts := newExporterOptions(options...)
+	if opts.recordMetrics {
+		dataPusher = dataPusher.withMetrics()
+	}
+
+	if opts.recordTrace {
+		spanName := config.Name() + spanNameSuffix
+		dataPusher = dataPusher.withSpan(spanName)
+	}
+
+	// The default shutdown function does nothing.
+	if opts.shutdown == nil {
+		opts.shutdown = func() error {
+			return nil
+		}
+	}
+
+	return &otlpTraceExporter{
+		exporterFullName: config.Name(),
+		dataPusher:       dataPusher,
+		shutdown:         opts.shutdown,
+	}, nil
+}
+
+// withMetrics wraps the current pusher into a function that records the metrics of the
+// pusher execution.
+func (next otlpTraceDataPusher) withMetrics() otlpTraceDataPusher {
```

Comment:
nit: for the context of the wrappers, not sure if `next` as the name for the pusher instance makes sense. May be just pusher would be better.